swaybar/event_loop.c

```diff
@@ -118,11 +112,29 @@ void event_loop_poll() {
 		struct pollfd pfd = event_loop.fds.items[i];
 		struct event_item *item = (struct event_item *)event_loop.items->items[i];
 
-		if (pfd.revents & pfd.events) {
+		// Always send these events
+		unsigned events = pfd.events | POLLHUP | POLLERR;
```

Comment:
-1

Reply:
This follows the semantics of poll. You get those events, even if you don't ask for them.
That's actually what's causing the issue.

Reply:
Ah.